xbmc/video/VideoItemArtworkHandler.h

```diff
@@ -0,0 +1,41 @@
+/*
+ *  Copyright (C) 2005-2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+class CFileItem;
+class CMediaSource;
+
+namespace VIDEO
+{
+class IVideoItemArtworkHandler
+{
+public:
+  static std::unique_ptr<IVideoItemArtworkHandler> Create(const std::shared_ptr<CFileItem>& item,
```

Comment:
this probably can be decoupled. Looks like a factory method outside of the scope of the Handler itself. Maybe create a factory class that returns the specific implementation?

Reply:
Could be done , yes, but what would be the benefit of this? Another new header / cpp file for the factory would make sense for this. To use it you actually will need to include both, handler and factory... not sure this is an improvement.

Reply:
The factory can be defined in the existing handler files. Would just decouple code and better identify component responsibilities but you can leave it as is if you feel it doesn't justify.
Personally I think it's cleaner and reads better for someone not familiar with our (absence of) architecture

Reply:
Fair point. I will change that.